utilities/general.py

```diff
@@ -331,3 +331,46 @@ def generate_random_name(prefix: str = "", length: int = 8) -> str:
     # random_uuid.hex is 32 characters long.
     suffix = random_uuid.hex[:length]
     return f"{prefix}-{suffix}" if prefix else suffix
+
+
+@retry(wait_timeout=Timeout.TIMEOUT_15_SEC, sleep=1)
+def wait_for_container_status(pod: Pod, container_name: str, expected_status: str) -> bool:
+    """
+    Wait for a container to be in the expected status.
+
+    Args:
+        pod: The pod to wait for
+        container_name: The name of the container to wait for
+        expected_status: The expected status
+
+    Returns:
+        bool: True if the container is in the expected status, False otherwise
+
+    Raises:
+        ResourceValueMismatch: If the container is not in the expected status
+    """
+
+    container_status = None
+    for cs in pod.instance.status.get("containerStatuses", []):
+        if cs.name == container_name:
+            container_status = cs
+            break
+    if container_status is None:
+        raise ResourceValueMismatch(f"Container {container_name} not found in pod {pod.name}")
+
+    if container_status.state.waiting:
+        reason = container_status.state.waiting.reason
+    elif container_status.state.terminated:
+        reason = container_status.state.terminated.reason
+    elif container_status.state.running:
+        # Running container does not have a reason
+        reason = "Running"
```

Comment:
if the container is running, we should exclude it from the check here.

Reply:
we still want to check that is running